lib/HLSL/HLOperationLower.cpp

```diff
@@ -2064,25 +2064,26 @@ Value *TrivialDotOperation(OP::OpCode opcode, Value *src0,
   return dotOP;
 }
 
-Value *TranslateIDot(Value *arg0, Value *arg1, unsigned vecSize, hlsl::OP *hlslOP, IRBuilder<> &Builder) {
+Value *TranslateIDot(Value *arg0, Value *arg1, unsigned vecSize, hlsl::OP *hlslOP, IRBuilder<> &Builder, bool isSigned = true) {
```

Comment:
Would prefer a bool representing unsigned (default to false) and comparison below against IOP_umul, since that's consistent with the pattern elsewhere.

Reply:
updated.